ray-curator/ray_curator/stages/text/embedders/base.py

```diff
@@ -0,0 +1,142 @@
+import os
+from dataclasses import dataclass
+from typing import Literal
+
+import cudf
+import cupy as cp
+import pandas as pd
+import torch
+import torch.nn.functional as F  # noqa: N812
+from crossfit.backend.cudf.series import create_list_series_from_1d_or_2d_ar
```

Comment:
I guess we can't remove the dependency for now?

Reply:
Good call, copied the code here